modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleServiceIT.java

```diff
@@ -476,7 +422,8 @@ public void testAutomaticForceMerge() throws Exception {
         int finalGeneration = randomIntBetween(3, 4);
         for (int currentGeneration = 1; currentGeneration < finalGeneration; currentGeneration++) {
             // This is currently the write index, but it will be rolled over as soon as data stream lifecycle runs:
-            final String toBeRolledOverIndex = getBackingIndices(dataStreamName).get(currentGeneration - 1);
+            final var backingIndexNames = waitForDataStreamBackingIndices(dataStreamName, currentGeneration);
```

Comment:
Were there failures here and you added the wait?

Reply:
Yep, https://github.com/elastic/elasticsearch/issues/124846 failed on that exact line. My theory of what happened is that at the end of this for-loop we fetch the data stream backing indices and we do so too at the start of this for-loop (i.e. this line). That's the same pattern we're seeing in the other test failures too.